Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -827,12 +827,26 @@ void ContainerNode::removeChildren()
 
     Ref protectedThis { *this };
     NodeVector removedChildren;
-    removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren);
-
+#if PLATFORM(MAC)
+    if (this != &rootNode() && this->document().settings().opportunisticallyDeleteNodes() && this->document().settings().opportunisticSweepingAndGarbageCollectionEnabled()) {
+        removeAllChildrenWithScriptAssertion(ChildChange::Source::API, removedChildren, DeferChildrenChanged::No, CanDelayNodeDeletion::Yes);
+        delayDeletingRemovedChildren(WTFMove(removedChildren));
```

Comment:
I think the place where we want to add this optimization is `removeDetachedChildrenInContainer`.
Instead of creating a `Vector<Ref<Node>>`, we can make each DOM node refer to the next node to delete using the next sibling pointer. That's a code we used to have prior to https://commits.webkit.org/254044@main (although it had become dead code by then).

Reply:
Actually, we probably need to add code here as well.
But we should really experiment with linked-list variant.
It was more efficient than using Vector at some point in the past.